### PR TITLE
Main menu overhaul (Part 2): simplify world creation

### PR DESCRIPTION
**Pull request by dseguin**
#### Summary
None

#### Purpose of change
- Contributes to #57098

([Also something that Erk outlined in a separate document](https://gist.github.com/I-am-Erk/d0b1d545dace7288962328061ca5e3f2))

> Option sliders in 'create new world' beginner settings. Each of these should be displayed as a slider, with a description to the right. So, for example:
>```
>Cities:
>----|---- Suburbia
>Difficulty:
>|-------- Cakewalk?
>Random NPCs:
>--------| Crowded
>```

#### Describe the solution
The world creation menu has been simplified into a single window (with the option of accessing standard world settings and mod menus):

https://user-images.githubusercontent.com/12537966/177019961-c5ae49b9-eec0-46fb-b38c-54be4afedf15.mp4

The initial screen contains a few basic slider options for adjusting world settings. These sliders are defined in JSON using the `option_slider` type. In the background, these sliders modify the standard settings depending on the selected level.

If the player chooses to customize settings via the advanced settings menu, the sliders are locked as "Custom". They can be accessible again by resetting the world options.

The menus allow basic mouse selection/navigation.

Additional changes:
- The "Finalize World" tab is gone, it was repurposed for this new menu.
- The world name no longer uses an embedded `string_input_popup`. Selecting the world name text displays a separate `string_input_popup` window for editing the text.
- The top tabs in the original world creation menu are still present while copying world settings, but the "Finalize World" tab was removed and instead a `string_input_popup` handles editing the world name.

#### Describe alternatives you've considered
I'm not 100% set on the final layout.

#### Testing

<details>
<summary>Minimum window size:</summary>

There's enough room to comfortably display all the info at the smallest window size (25x80):

small

</details>

<details>
<summary>Content scrolling:</summary>

If there are more sliders than what can be displayed on screen, the screen can be scrolled by navigating up and down the options:

https://user-images.githubusercontent.com/12537966/177019956-9fca50fb-721a-4257-93fb-beb742428a18.mp4

</details>

<details>

<summary>Copy World Settings:</summary>

The "Copy World Settings" still shows the first two tabs, and the final tab is replaced by a string input popup for choosing the new world's name:

copy_world
copy_world_tabs

</details>

<details>
<summary>Curses build:</summary>

The curses build works exactly the same, just without SDL mouse support:

https://user-images.githubusercontent.com/12537966/176815658-d0350f7d-5fcf-4a73-981d-81cef29e72e3.mp4

</details>

#### Additional context
TODO:
- [x] Reorganize world creation menus
- [x] Implement basic option sliders
- [x] Implement basic mouse navigation
- [x] JSONify option sliders
  - [x] Documentation
  - [x] Unit tests
- [x] Make the primary window scrollable (in case more sliders are added in the future)

### CODE REVIEW

File: src/worldfactory.cpp [hunk 1]
```diff
@@ -1495,136 +1531,409 @@ int worldfactory::show_worldgen_tab_modselection( const catacurses::window &win,
...
+                                      .text( worldname )
+                                      .query_string();
+                if( !popup.canceled() ) {
+                    world->world_name = worldname = message;
+                }
+            } else if( sel_opt == static_cast<int>( sliders.size() + 1 ) ) {
+                // finish
+                if( worldname.empty() ) {
+                    noname = true;
+                    ui_manager::redraw();
```
Comment (Qrox) [inline src/worldfactory.cpp:1857]:
```suggestion
                    ui.invalidate_ui();
```

`query_yn` also calls `ui_manager::redraw()` so simply invalidating the UI should work.

**Comment by PatrikLundell** [comment]
The keyboard supports seems to be lacking, i.e. no indications of which keys you need to use either to move between sliders or move the sliders themselves, and there's no key binding menu indication, so you can't look it up either.

**Comment by snipercup** [comment]
Awesome! Can you make a control to have the sliders jump to random positions? I always play random but the world options didn't have this feature.

**Comment by Night-Pryanik** [comment]
World name is now colored magenta on black. This puts a tension on my eyes. Maybe change it to default white on black? Or make it more contrast, like yellow on black?

File: src/worldfactory.cpp [hunk 2]
```diff
@@ -158,41 +157,61 @@ WORLDPTR worldfactory::make_new_world( bool show_prompt, const std::string &worl
...
+    }
+    if( curtab >= 0 ) {
+        string_input_popup str_input;
+        std::string nname = str_input
+                            .max_length( max_worldname_len )
+                            .title( _( "Choose a new name for this world." ) )
+                            .text( world->world_name )
+                            .query_string();
+        if( str_input.canceled() || nname.empty() ) {
+            // use the same name?
```
Comment (Night-Pryanik) [inline src/worldfactory.cpp:207]: Print `________NO NAME ENTERED!________`.

**Comment by descan** [comment]
A minor point, but considering how fairly regularly people play in Megacities, having the highest tier on density be a megacity (max size of cities, 2 or so distance) and labeled as such could be a nice nod.

**Comment by dseguin** [comment]
Cheers for the feedback!

From @Night-Pryanik:
- Various text changes
- Changed world name color to bright pink to contrast against the dark background
- Made "Advanced options" and "Open mod manager" accessible via mouse click

From @snipercup:
- Added a `Randomize` button to randomize the slider options

From @descan:
- Added "Megacity" and "Remote" tiers that represent the maximum and minimum size and proximity for cities

From @PatrikLundell:
- Added text hint for keyboard navigation

changes2

**Pull request merged**